torch/testing/_internal/common_methods_invocations.py

```diff
@@ -7290,6 +7290,69 @@ def sample_inputs_dropout_backward(op_info, device, dtype, requires_grad, **kwar
     for case, scale in product(cases, scale_vals):
         yield SampleInput(make_arg(case), make_mask(case), scale)
 
+def sample_inputs_embedding_bag_dense_backward(op_info, device, dtype, requires_grad, **kwargs):
+    for sample in sample_inputs_embedding_bag(op_info, device, dtype, requires_grad, **kwargs):
+        weight = sample.input  # NB: is flipped with indices!
+        assert len(sample.args) == 1
+        indices = sample.args[0]
+        offsets = sample.kwargs.pop('offsets', None)
+        per_sample_weights = sample.kwargs.pop('per_sample_weights', None)
+        # Some of the logic here is cribbed from torch.nn.functional.embedding_bag
+        if offsets is None:
+            offsets = torch.arange(0, indices.numel(), indices.size(1), dtype=indices.dtype, device=indices.device)
+            indices = indices.reshape(-1)
+            if per_sample_weights is not None:
+                per_sample_weights = per_sample_weights.reshape(-1)
+        num_weights = weight.size(0)
+        scale_grad_by_freq = sample.kwargs.pop('scale_grad_by_freq', False)
+        mode = sample.kwargs.pop('mode', 'mean')
+        if mode == "sum":
+            mode_enum = 0
+        elif mode == "mean":
+            mode_enum = 1
+        elif mode == "max":
+            mode_enum = 2
+        padding_idx = sample.kwargs.pop('padding_idx', None)
+        include_last_offset = sample.kwargs.pop('include_last_offset', False)
+
+        if padding_idx is None:
+            padding_idx = -1
+        else:
+            from torch._meta_registrations import maybe_wrap_dim
+            padding_idx = maybe_wrap_dim(padding_idx, weight.size(0))
+
+        sample.kwargs.pop('max_norm', None)  # not used by the op
+        sample.kwargs.pop('norm_type', None)  # not used by the op
+
+        # This is a precondition for the function, so force it
```

Comment:
Isn't this already True if it is a precondition for the forward?

Reply:
No, not precondition for forward. The backward only gets called in contexts where the internal private function was guaranteed to be contiguous

Reply:
Ok ok